src/types/bearing_nb.cpp

```diff
@@ -0,0 +1,24 @@
+#include "engine/bearing.hpp"
+
+#include <nanobind/nanobind.h>
+#include <nanobind/operators.h>
+#include <nanobind/stl/pair.h>
+
+namespace nb = nanobind;
+
+void init_Bearing(nb::module_& m) {
+    using osrm::engine::Bearing;
+
+    nb::class_<Bearing>(m, "Bearing")
+        .def(nb::init<>())
+        .def("__init__", [](Bearing* t, std::pair<short, short> pair) {
+            new (t) Bearing{pair.first, pair.second};
+         })
+        .def_rw("bearing", &Bearing::bearing)
+        .def_rw("range", &Bearing::range)
+        .def("IsValid", &Bearing::IsValid)
+        .def(nb::self == nb::self)
```

Comment:
what this `nb::self == nb::self` means?

Reply:
Those are operator bindings, so it would allow `==` comparisons for the Bearing object.